xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -1154,12 +1154,26 @@ void CActiveAESink::GenerateNoise()
 
 void CActiveAESink::SetSilenceTimer()
 {
-  if (m_extStreaming)
-    m_extSilenceTimeout = std::chrono::milliseconds::max();
-  else if (m_extAppFocused)
-    m_extSilenceTimeout = m_silenceTimeOut;
+  if (m_extStreaming) // handles playback
+  {
+    // Limit to 3h to avoid overflows.
+    // In any case, this condition (timeout) will never occur: 3 hours of playback would be
+    // necessary but *without sending audio*, which makes no sense.
+    // During playback the timer is constantly reset to zero because audio is being sent
+    m_extSilenceTimeout = 3h;
```

Comment:
If this overflows somewhere we should handle that rather than try and limit it here.

Reply:
This would even work fine putting 10s of timeout. What I'm trying to explain is that an infinite timer is not necessary.

Reply:
If that's the case the comment doesn't make any sense and I would recommend removing it.

Reply:
Remove this "state" can cause unexpected side effects and timeout needs to be > 0ms

Reply:
It's still not clear to me what this actually fixes? Before the use of `std::chrono` it was set to `std::numeric_limits<unsigned int>::max()`.

Reply:
Prevents "possibles" overflows to negative values as std::chrono can have negative values:

![not-initialized](https://user-images.githubusercontent.com/58434170/210013596-18406a3f-fe27-49a1-b0c9-76f4fed6603d.png)

before chrono changes this was **unsigned** int